add-api/client/src/components/GitHubAuthButton.tsx

```diff
@@ -0,0 +1,22 @@
+// GitHubAuthButton.tsx
```

Comment:
No all of these permissions are unnecessary, I will updates this. And yea, Gorilla can take ownership, [just create a new OAuth App ](https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/creating-an-oauth-app)and send me the Client ID and Client Secret, I can update the server and frontend with these new values.

Reply:
I was only able to reduce the scope of the OAuth app down to public repos. However, within this scope, the app is still authorized to perform read and write operations on:

- Code
- Issues
- Pull requests
-Wikis
- Settings
- Webhooks and services
- Deploy keys

Unfortunately, OAuth apps do not support more granular permissions for these; these permissions are bundled together with access to public repos by GitHub’s design. To address this, we could register a GitHub App for the Gorilla in the future, which has more granular permission controls, and rewrite the server to work with the Github App instead of OAuth